pkg/platform/runtime/runtime.go

```diff
@@ -141,15 +141,32 @@ func (r *Runtime) validateCache() error {
 
 	// Check if local build script has changes that should be committed.
 	if r.cfg.GetBool(constants.OptinBuildscriptsConfig) {
-		if cachedScript, err := r.store.BuildScript(); err == nil {
+		cachedScript, err := r.store.BuildScript()
+		switch {
+		case err == nil:
 			script, err := buildscript.ScriptFromProject(r.target)
-			if err != nil {
+			if err != nil && !errs.Matches(err, buildscript.ErrBuildscriptNotExist) {
 				return errs.Wrap(err, "Unable to get local build script")
 			}
-			if !script.Equals(cachedScript) {
+			if script != nil && !script.Equals(cachedScript) {
 				return &NeedsCommitError{errs.New("Runtime changes should be committed")}
 			}
-		} else if err != store.ErrNoBuildScriptFile {
+
+		case err == store.ErrNoBuildScriptFile:
```

Comment:
```suggestion
		case errors.Is(err, store.ErrNoBuildScriptFile):
```

`errors.Is()` will work even if `store.ErrNoBuildScriptFile` is not the top level error.